src/mimefactory.rs

```diff
@@ -793,7 +835,7 @@ impl<'a> MimeFactory<'a> {
 
             message
         } else {
-            // Store hidden headers in the inner unencrypted message.
+            // Store hidden headers in the outer message.
```

Comment:
Maybe remove these comments at all? If it's a signed-only message, hidden headers go to the signed (inner) part